src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs

```diff
@@ -224,10 +228,58 @@ public async Task<List<SubmittedApplicationAnswer>> ExtractAnswersForApplication
                     }
                 }
             }
+            //Extract QuestionId Y0-110 answer even if it is inactive, there is an issue setting answer active to false while adding orgnisation type answer to the tabular list as the last entry
+            var hasPartnershipAnswers = answers.Any(answer => answer.QuestionId == QuestionIdPartnership);
+            if(!hasPartnershipAnswers)
+            {
+                await ExtractPartnershipAnwers(applicationId, answers);
+            }
 
             return answers;
         }
 
+        private async Task ExtractPartnershipAnwers(Guid applicationId, List<SubmittedApplicationAnswer> answers)
+        {
+            try
+            {
+                var partnershipAnwers = await ExtractAnwersByQuestionTag(applicationId, AddPartners, QuestionIdPartnership);
+                if (partnershipAnwers!= null)
+                {
+                    var tabularData = JsonConvert.DeserializeObject<TabularData>(partnershipAnwers.Value);
+
+                    var question = new Question
+                    {
+                        QuestionId = QuestionIdPartnership,
+                        Input = new Input()
+                    };
+                    question.Input.Type = TabularDataType;
+                    var partnershipTabularAnswers = TabularDataMapper.GetAnswers(applicationId, YourOrganisation, WhosInControl, PageIdPartnershipAddPartners, question, tabularData);
+                    answers.AddRange(partnershipTabularAnswers);
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, $"Unable to extracted Partnership anwers for Application: {applicationId}");
+            }
+        }
+
+        private async Task<Answer> ExtractAnwersByQuestionTag(Guid applicationId, string questionTag, string questionId = null)
+        {
+            var answer = new Answer { QuestionId = questionId };
+
+            var questionTagData = await _qnaApiClient.GetTabularDataByTag(applicationId, questionTag);
+            if (questionTagData != null)
```

Comment:
I am not able to follow this, can this be simplified as... 
```
if (questionTagData == null) return null;
answer.Value = questionTagData;
return answer;
```

Reply:
Refactored this